File: HashTable/1133_Largest_Unique_Number/1133_Largest_Unique_Number.java
import java.util.HashMap;
import java.util.Collections;
import java.util.ArrayList;

public class Solution {
    public int largestUniqueNumber(int[] A) {
        HashMap<Integer, Integer> countMap = new HashMap<>();
        for (int num : A) {
            countMap.put(num, countMap.getOrDefault(num, 0) + 1);
        }

        ArrayList<Integer> uniqueNums = new ArrayList<>();
        for (int key : countMap.keySet()) {
            if (countMap.get(key) == 1) {
                uniqueNums.add(key);
            }
        }

        if (uniqueNums.isEmpty()) {
            return -1;
        }

        Collections.sort(uniqueNums);
        return uniqueNums.get(uniqueNums.size() - 1);
    }
}

File: HashTable/1133_Largest_Unique_Number/1133_Largest_Unique_Number.cpp
#include <vector>
#include <unordered_map>
#include <algorithm>
using namespace std;

class Solution {
public:
    int largestUniqueNumber(vector<int>& A) {
        unordered_map<int, int> countMap;
        for (int num : A) {
            countMap[num]++;
        }

        vector<int> uniqueNums;
        for (auto& it : countMap) {
            if (it.second == 1) {
                uniqueNums.push_back(it.first);
            }
        }

        if (uniqueNums.empty()) {
            return -1;
        }

        sort(uniqueNums.begin(), uniqueNums.end());
        return uniqueNums.back();
    }
};
